Library/Homebrew/download_strategy.rb

```diff
@@ -488,7 +488,12 @@ def resolve_url_basename_time_file_size(url, timeout: nil)
     @resolved_info_cache ||= {}
     return @resolved_info_cache[url] if @resolved_info_cache.include?(url)
 
-    parsed_output = curl_headers(url.to_s, wanted_headers: ["content-disposition"], timeout: timeout)
+    begin
+      parsed_output = curl_headers(url.to_s, wanted_headers: ["content-disposition"], timeout: timeout)
+    rescue ErrorDuringExecution
+      return [url, parse_basename(url), nil, nil, false]
```

Comment:
Required by your change but just a drive-by thought that this is a _horrendous_ return signature that's dying to be a e.g. `Hash` or `Struct`.

Reply:
Yeah, agreed. Takes me a moment to understand everytime I look at it.

It's also difficult to change without breaking custom download strategies, so probably worth doing with a larger refactor with better public API.